app/code/Magento/CheckoutAgreements/Model/ResourceModel/Agreement/Grid/Collection.php

```diff
@@ -64,7 +64,7 @@ private function getStoresForAgreements()
 
         if (!empty($agreementId)) {
             $select = $this->getConnection()->select()->from(
-                ['agreement_store' => 'checkout_agreement_store']
+                ['agreement_store' => $this->getResource()->getTable('checkout_agreement_store')]
```

Comment:
thank you for your fix. The solution looks reasonable. The reason why this issue appeared is that there is not an integration test for this case. We are running integration tests with and w/o table prefixes to make sure that such issues won't appear. 

Please cover this case with the integration test (and earn additional contribution points :) ) to avoid a regression issue for your solution.
Thank you

Reply:
@okobchenko is `getResource()` really neccessary?